Reference/Common-Pitfalls/index-v9.md

```diff
@@ -0,0 +1,424 @@
+
+This means that there is now a minimum of __20,000__ new objects created and allocated in memory. The number of traversals/visits to each
+of these objects is now 5000 x 5000 = __25,000,000 (25 MILLION TIMES!)__
+
+_Side note: The other problem is the logic used to lookup related recipes is incredibly inefficient. Instead, each recipe should have a picker to choose its related recipe's and then each of those can be looked up by their ID.
+(There's probably a few other ways to achieve this too!)_
+
+This leads us on to the next anti-pattern...
+
+## Don't eager load data, lazy load it instead
+
+The above example could be rewritten like this:
+
+```csharp
+    public class RecipeModel : PublishedContentWrapped
+    {
+        public RecipeModel(IPublishedContent content, IPublishedValueFallback publishedValueFallback) : base(content, publishedValueFallback)
+        {}
+
+        private int? _votes;
+        public int Votes
+        {
+            get
+            {
+                // Lazy load the property value and ensure it's not re-resolved once it's loaded
+                return _votes ??= this.Value<int>("votes");
+            }
+        }
+
+        // Just return the Ids, they can be resolved to IPublishedContent instances in the view or elsewhere,
+        // doesn't need to be in the model - this would also be bad if the model was cached since all of the
+        // related entities would end up in the cache too.
+        private List<int> _related;
+
+        public IEnumerable<int> RelatedRecipes => _related ??= this.Value<IEnumerable<int>>("related").ToList();
+    }
+```
+
+This is slightly better:
+
+* This will iterate over all Recipes, create and allocate 5000 instances of `IPublishedContent`
+* This will create and allocate 5000 instances of `RecipeModel`
+
+This means that there is now a minimum of __15,000__ new objects created and allocated in memory. The number of traversals/visits to each
+of these objects is now __5000__.
+
+This is still not great though. There really isn't much reason to create a `RecipeModel` to use it as a filter,
+this is allocating a lot of objects to memory for no real reason. This could be written like:
+
+```csharp
+@{
+	var recipeNode = Umbraco.Content(1234);
+}
+
+<ul>
+	@foreach (var recipe in recipeNode.Children
+		.OrderByDescending(x => x.Value<int>("votes"))
+		.Take(10))
+	{
+		<li><a href="@recipe.Url()">@recipe.Name</a></li>
+	}
+</ul>
+```
+
+This is slightly better:
+
+This means that there is now a minimum of __10,000__ new objects created and allocated in memory. The number of traversals/visits to each
+of these objects is now __5000__.
+
+## Not caching expensive lookups
+
+Based on the above 2 points, you can see that iterating content with the traversal APIs will cause new instances of `IPublishedContent` to be created. When memory is used, Garbage Collection needs to occur and this
```

Comment:
As above, this will no longer allocate new instances. But this concept still stands that the more you allocate the more the GC needs to clean up